lib/factory.js

```diff
@@ -15,8 +15,15 @@ export function createSigner({id, secretKey}) {
     algorithm: curve,
     id,
     async sign({data} = {}) {
-      return new Uint8Array(await webcrypto.subtle.sign(
-        algorithm, secretKey, data));
+      if(curve === ECDSA_CURVE.secp256k1) {
+        const {Crypto} = await import('@peculiar/webcrypto');
```

Comment:
We will have to figure something else out here, I don't expect us to pull in this whole library nor list it as a dependency. I'm not sure what static analysis will do here either, even though this is "just in time" / dynamic imported. We have to think of this multikey library being included in places where the additional `@peculiar/webcrypto` dependency isn't needed and the security/auditing overhead/issues it creates. cc: @mattcollier @davidlehn

Instead, we could possibly allow for some API (interface) injection via a param where the key is imported/generated or an additional optional registration (aka `use()`) API as setup. Either that or just require the caller to have polyfilled this and provide some docs on how to do it. A proper polyfill installation would also remove all of the extra optional and duplicated code below (I'd hope).

Reply:
Yeah I agree, 

However it has been stated that there is no intent to support secp256k1 in the `webcrypto` node implementation (not sure if I can find that conversation again).

DI would be good as a per customer basis, but the thing is it's not just about importing the library, it's also consuming it instead of the native `webcrypto`. I think there would be a bigger vulnerability risk by allowing injection of a webcrypto polyfill (which could overwrite any curve) than by fully controlling where, when and how it's used by a dynamic require.

Dynamic import in my experience is quite smart about being there but only used as needed, so not polluting other cases.